### PR TITLE
Initial support for OpenBSD

### PR DESCRIPTION
**Pull request by qbit**
Unfortunately we currently lack libunwind (work is being done on it), so we can't get to far into building. This helps get us a bit further!

### CODE REVIEW

File: src/pal/tools/gen-buildsys-clang.sh [hunk 1]
```diff
@@ -33,8 +33,8 @@ OS=`uname`
 # On FreeBSD the version number is appended without point and dash (i.e.
 # llvm-ar35).
 # Additionally, OSX doesn't use the llvm- prefix.
-if [ $OS = "Linux" -o $OS = "FreeBSD" ]; then
-  llvm_prefix="llvm-"
+if [ $OS = "Linux" -o $OS = "FreeBSD" -o $OS = "OpenBSD" ]; then
+    llvm_prefix="llvm-"
```
Comment (jkotas) [inline src/pal/tools/gen-buildsys-clang.sh]: Could you please make the indentation match the surrounding code?

**Comment by jkotas** [comment]
LGTM, modulo comment. Could you please squash the commit history?

**Comment by qbit** [comment]
squashed!

**Comment by jkotas** [comment]
Thanks for the contribution!

**Pull request merged**